experimental/packages/shim-opencensus/src/metric-transform.ts

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as oc from '@opencensus/core';
+import { Attributes, HrTime, ValueType, diag } from '@opentelemetry/api';
+import {
+  AggregationTemporality,
+  DataPoint,
+  DataPointType,
+  GaugeMetricData,
+  HistogramMetricData,
+  InstrumentType,
+  MetricData,
+  SumMetricData,
+} from '@opentelemetry/sdk-metrics';
+import { FlatMap } from '@opentelemetry/sdk-metrics/src/utils';
+
+type BaseMetric = Omit<MetricData, 'dataPoints' | 'dataPointType'>;
+interface MappedType {
+  type: InstrumentType;
+  valueType: ValueType;
+  dataPointType:
+    | DataPointType.GAUGE
+    | DataPointType.SUM
+    | DataPointType.HISTOGRAM;
+}
+const ZEROED_HRTIME: HrTime = [0, 0];
+
+export function mapOcMetric(metric: oc.Metric): MetricData | null {
+  const { description, name, unit, type } = metric.descriptor;
+  const mappedType = mapOcMetricDescriptorType(type);
+  if (mappedType === null) {
+    return null;
+  }
+
+  const baseMetric: BaseMetric = {
+    aggregationTemporality: AggregationTemporality.CUMULATIVE,
+    descriptor: {
+      description,
+      name,
+      unit,
+      type: mappedType.type,
+      valueType: mappedType.valueType,
+    },
+  };
+
+  switch (mappedType.dataPointType) {
+    case DataPointType.GAUGE:
+      return gauge(metric, mappedType.dataPointType, baseMetric);
+    case DataPointType.SUM:
+      return sum(metric, mappedType.dataPointType, baseMetric);
+    case DataPointType.HISTOGRAM:
+      return histogram(metric, mappedType.dataPointType, baseMetric);
+  }
+}
+
+function mapOcMetricDescriptorType(
+  type: oc.MetricDescriptorType
+): MappedType | null {
+  switch (type) {
+    case oc.MetricDescriptorType.GAUGE_INT64:
+      return {
+        type: InstrumentType.OBSERVABLE_GAUGE,
```

Comment:
How do you know whether the instrument was observable or not (e.g. for counters)?

Reply:
Afaict from OpenCensus code, there is no way to tell. I'm receiving a list of OpenCensus `Metric` and here is the definitions https://github.com/census-instrumentation/opencensus-node/blob/d46c8891b15783803d724b717db9a8c22cb73d6a/packages/opencensus-core/src/metrics/export/types.ts#L18

Do the assumptions here seem reasonable?

Reply:
Actually, is type even used anywhere?  It looks like gauge(), sum(), and histogram() all use dataPointType instead.

Reply:
It is returned from the producer as part of MetricData. See L54

Reply:
ah, got it.  Thanks.  I think this is fine then.

Reply:
The field `type` is going to be deprecated in https://github.com/open-telemetry/opentelemetry-js/pull/3520. It doesn't make sense to be exposed to metric readers and exporters. So assuming it is an observable should be fine here.